helix-term/src/compositor.rs

```diff
@@ -27,6 +27,24 @@ pub struct Context<'a> {
     pub jobs: &'a mut Jobs,
 }
 
+impl<'a> Context<'a> {
+    /// Waits on all pending jobs, and then tries to flush all pending write
+    /// operations for all documents.
+    pub fn block_try_flush_writes(&mut self) -> anyhow::Result<()> {
+        tokio::task::block_in_place(|| {
+            helix_lsp::block_on(self.jobs.finish(Some(self.editor), None))
```

Comment:
This has a timeout, right?

Reply:
Currently, this is only controlled by each future/job. For example, the LSP client always wraps the requests in a timeout. But there is no globally enforced timeout. This is the case as of today as well.